web/packages/teleport/src/Navigation/SideNavigation/Section.tsx

```diff
@@ -18,46 +18,148 @@
 
 import React from 'react';
 import { NavLink } from 'react-router-dom';
-import styled, { css } from 'styled-components';
+import styled, { css, useTheme } from 'styled-components';
 
-import { Box, ButtonIcon, Flex, Text } from 'design';
+import { Box, ButtonIcon, Flex, P2, Text } from 'design';
 import { Theme } from 'design/theme';
 import { ArrowLineLeft } from 'design/Icon';
-import { HoverTooltip } from 'shared/components/ToolTip';
+import { HoverTooltip, ToolTipInfo } from 'shared/components/ToolTip';
 
-import { NavigationSection } from './Navigation';
+import cfg from 'teleport/config';
+
+import { NavigationSection, NavigationSubsection } from './Navigation';
 import { zIndexMap } from './zIndexMap';
 import { CategoryIcon } from './CategoryIcon';
 
-export function Section({
-  section,
-  $active,
-  setExpandedSection,
-  onClick,
-  children,
-  isExpanded,
-}: {
+type SharedSectionProps = {
   section: NavigationSection;
   $active: boolean;
-  setExpandedSection: () => void;
-  onClick: (event: React.MouseEvent) => void;
-  isExpanded?: boolean;
-  children?: JSX.Element;
-}) {
+  isExpanded: boolean;
+  setExpandedSection: (section: NavigationSection) => void;
+};
+
+/**
+ * CustomChildrenSectionProps is a type for a NavigationSection with custom children (e.g. search section).
+ */
+type CustomChildrenSectionProps = SharedSectionProps & {
+  type: 'custom-children';
+  children: JSX.Element;
+};
+
+/**
+ * DefaultSectionProps is a type for a NavigationSection with default children, this will just be a list of subsection items (eg. Identity section).
+ */
+type DefaultSectionProps = SharedSectionProps & {
+  type: 'default-children';
+  currentView?: NavigationSubsection;
+  handleNavigation?: (route: string) => void;
+  currentPageSection?: NavigationSection;
+  stickyMode: boolean;
+  toggleStickyMode: () => void;
+  previousExpandedSection: NavigationSection;
+};
+
+export function Section(
```

Comment:
What's the benefit of having a single component with conditional props? 
I believe it would be simpler to split it into two components: `CustomChildrenSection` and `DefaultSection`.